google/cloud/internal/curl_impl.cc

```diff
@@ -655,7 +655,8 @@ StatusOr<std::size_t> CurlImpl::ReadImpl(absl::Span<char> output) {
   if (curl_closed_) {
     OnTransferDone();
     status = google::cloud::rest_internal::AsStatus(
-        static_cast<HttpStatusCode>(http_code_), {});
+        static_cast<HttpStatusCode>(http_code_),
+        std::string(spill_.data(), spill_offset_));
```

Comment:
There is no guarantee that the spill buffer has the error data.  Nor that it is large enough to hold all the error data.